src/internal/pfsdb/commits.go

```diff
@@ -196,6 +196,13 @@ func CreateCommit(ctx context.Context, tx *pachsql.Tx, commitInfo *pfs.CommitInf
 	if err := validateCommitInfo(commitInfo); err != nil {
 		return 0, err
 	}
+	commit, err := GetCommitWithIDByKey(ctx, tx, commitInfo.Commit)
+	if err == nil && commit != nil {
+		return 0, &CommitAlreadyExistsError{CommitID: commit.CommitInfo.Commit.Key()}
+	}
+	if err != nil && (errors.As(err, new(*ProjectNotFoundError)) || errors.As(err, new(*RepoNotFoundError))) {
```

Comment:
It would be nice if we made errors.Is work for this.  I know that the default behavior is that `ProjectNotFoundError{Project: a}` `Is`n't `ProjectNotFoundError{Project: b}` but we should probably adjust that unless we rely on the behavior elsewhere.  I think using `As` to mean `Is` is strange (i.e. discarding the converted matched error as this code does).

Reply:
I like the pattern where this is provided by a `func IsErrProjectNotFound(err error) bool` in the same package.  `IsErrProjectNotFound` is implemented in terms of `errors.As`.

Reply:
Separately: I don't know what tool told me this, but something encouraged me to give error types names that start with `Err`.

Reply:
that's govet (more likely) or staticcheck (less likely).  both are built into gopls.